include/onnxruntime/core/common/hash_value.h

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cstdint>
+
+namespace onnxruntime {
+
+/** A computed hash value. */
+using HashValue = uint64_t;
```

Comment:
Would HashValue_t or HashValueType be better to convey that this is a type?

Reply:
some other examples to consider for precedent:
https://github.com/microsoft/onnxruntime/blob/9d9ebd3b8536cc93fe8ac32acdffcd490c16c70c/include/onnxruntime/core/graph/basic_types.h#L27-L32

given a name like "NodeIndex", "HashValue" seemed reasonable to me

Reply:
Ok, looks like we're inconsistent given that we've ProviderType as well.

Reply:
Just curious, why not put this into the `basic_types.h`, for simplicity?

Reply:
> Just curious, why not put this into the basic_types.h, for simplicity?

basic_types.h currently seems to be graph-specific. but perhaps I can add a more general one.

Reply:
moved HashValue to core/common/basic_types.h